src/main/java/org/sagebionetworks/bridge/user/BootstrapTestUser.java

```diff
@@ -0,0 +1,72 @@
+package org.sagebionetworks.bridge.user;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.sagebionetworks.bridge.rest.ClientManager;
+import org.sagebionetworks.bridge.rest.Config;
+import org.sagebionetworks.bridge.rest.model.Phone;
+import org.sagebionetworks.bridge.rest.model.Role;
+import org.sagebionetworks.bridge.rest.model.SignIn;
+import org.sagebionetworks.bridge.rest.model.UserSessionInfo;
+
+class BootstrapTestUser implements TestUser {
+    
+    private SignIn signIn;
+    private ClientManager manager;
+    private String userId; // try and hold onto this for the sake of cleaning up tests
+
+    public BootstrapTestUser(SignIn signIn, ClientManager manager, String userId) {
+        checkNotNull(signIn.getAppId());
+        this.signIn = signIn;
+        this.manager = checkNotNull(manager);
+        this.userId = userId; // if this is null, we will try and get it on a sign in
+    }
+    public UserSessionInfo getSession() {
+        return manager.getSessionOfClients();
+    }
```

Comment:
A lot of these are identical to TestUserImpl. Would it be better to make TestUser an abstract class (possibly called AbstractTestUser) and have these methods be part of the base class? Then BootstrapTestUser and TestUserImpl only need to contain the methods they differ on.

Reply:
Made TestUser the base implementation and then overrode three of the methods in the Bootstrap class. Much cleaner.